### PR TITLE
fix(deployments): reappearing deployments continue updates

### PR DESCRIPTION
**Pull request by andrewazores**
This PR modifies the DeploymentsService timeseries Observables' logic so that deployments which "reappear", ex. gaining a deployment in a new environment (promotion to "run") or scaling up from 0 pods, receive timeseries stats updates for CPU, Memory, and Network usage. Prior to these patches there is a bug where the timeseries stats will never be updated in these situations, as documented [here](https://github.com/openshiftio/openshift.io/issues/3454).

This PR also includes a patch to ensure that the correct chart component object is updated when these events occur, otherwise it is possible for the first such updated series may set appearance properties on an outdated and invalid chart object.

### CODE REVIEW

**Comment by jiekang** [comment]
With this patch, when I expand the card, no request is made to acquire statistics until what looks like the next polling interval.

**Comment by andrewazores** [comment]
@jiekang thanks, addressed in latest commit.

File: src/app/space/create/deployments/apps/deployment-details.component.ts [hunk 1]
```diff
@@ -164,6 +160,16 @@ export class DeploymentDetailsComponent {
         .subscribe(this.hasPods)
     );
 
+    this.subscriptions.push(
+      this.hasPods.subscribe((hasPods: boolean): void => {
+        if (!hasPods) {
+          this.cpuChart.next(null);
```
Comment (jiekang) [inline src/app/space/create/deployments/apps/deployment-details.component.ts]: I've grown to not be a fan of using null as a state for some object. Is it possible to introduce an EMPTY_STATE that's null, or better yet an actual object that represents this?

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2910-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2910-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/apps/deployment-details.component.ts [hunk 2]
```diff
@@ -143,13 +143,11 @@ export class DeploymentDetailsComponent {
...
-  private memChart: ChartAPI;
-  private netChart: ChartAPI;
+  private cpuChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
+  private memChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
+  private netChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
 
-  private cpuChartLoad: Subject<void> = new ReplaySubject<void>();
-  private memChartLoad: Subject<void> = new ReplaySubject<void>();
-  private netChartLoad: Subject<void> = new ReplaySubject<void>();
+  private static NO_CHART: ChartAPI = {} as ChartAPI;
```
Comment (jiekang) [inline src/app/space/create/deployments/apps/deployment-details.component.ts]: I might actually prefer `null` over `{} as ChartAPI` as the latter can break the type assumption one has from function definitions. E.g.

```
  aFunction(a: boolean): X {
    if (a) { 
      return new X();
    }

    return {} as X;
  }
```

If X contains any non-optional properties, this function is very misleading to a user who doesn't read the implementation, which could involve many other inner functions. The more accurate definition would be
```
aFunction(a: boolean): X | {} {
  [...]
}
```

This increases the likelihood a user will expect and handle the empty object properly. I would try hard to not use `as` keyword outside of tests though.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2910-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2910-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2910-8 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2910-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2910-9 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2910-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2910-10 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2910-fabric8-ui.badger.fabric8.io

**Pull request merged**